app/gui/src/dashboard/modals/CreateCredentialModal.tsx

```diff
@@ -0,0 +1,45 @@
+/** @file A modal for creating and editing a credential. */
+import { Dialog, Dropdown, Text } from '#/components/AriaComponents'
+import { CREDENTIAL_INFOS } from '#/data/serviceCredentials'
+import { makeCredentialCreationHandler } from '#/data/serviceCredentials/logic'
+import { useText } from '#/providers/TextProvider'
+import type { CredentialConfig, SecretId } from '#/services/Backend'
+import { useState } from 'react'
+
+/** Props for a {@link CreateCredentialModal}. */
+export interface CreateCredentialModalProps {
+  readonly noDialog?: boolean
+  readonly doCreate: (name: string, value: CredentialConfig) => Promise<SecretId>
+}
+
+/** A modal for creating a credential. */
+export default function CreateCredentialModal(props: CreateCredentialModalProps) {
+  const { noDialog = false, doCreate } = props
+  const { getText } = useText()
+  const [selectedChildIndex, setSelectedChildIndex] = useState<number>(0)
+  const createCredentialsHandler = makeCredentialCreationHandler(doCreate)
+
+  const selectedItem = CREDENTIAL_INFOS[selectedChildIndex]
+  const content = (
+    <div className="w-full">
```

Comment:
I assume this should contain `Form` here?

Reply:
don't believe so - the form is in the nested/dynamic component, it's not type-safe otherwise (bc forms can't be nested)

Reply:
Yes, nested forms were causing lots of trouble here.